Gordon360/Authorization/StateYourBusiness.cs

```diff
@@ -686,7 +715,30 @@ private async Task<bool> CanUpdateAsync(string resource)
 
                         return false;
                     }
+                    
+                case Resource.RECIM_ACTIVITY:
+                case Resource.RECIM_SERIES:
+                case Resource.RECIM_SPORT:
+                    {
+                        return _participantService.IsAdmin(user_name);
+                    }
 
+                case Resource.RECIM_TEAM:
+                    {
+                        var teamID = (int)context.ActionArguments["teamID"];
+                        return _teamService.IsTeamCaptain(user_name, teamID) || _participantService.IsAdmin(user_name);
+                    }
+
+                case Resource.RECIM_MATCH:
+                    {
+                        if (context.ActionArguments["matchID"] is int matchID)
+                        {
+                        var match = _matchService.GetSimpleMatchViewByID(matchID);
+                        var series = _seriesService.GetSeriesByID(match.SeriesID);
+                        return _activityService.IsReferee(user_name, series.ActivityID) || _participantService.IsAdmin(user_name);
```

Comment:
> You could make this more efficient using navigation properties, assuming you used the database model instead of the DB model.
> 
> For example:
> 
> ```cs
>     var activityID = _context.Match.Find(matchID).Series.ActivityID;
>     return _activityService.IsReferee(user_name, activityID);
> ```

im assuming you meant db model instead of viewmodel?

Reply:
Regardless, it works! Once again, I did not set up SYB but I did notice CCTcontext and was wondering why we never used it. I'll most likely comb through SYB to truncate anything that could have just used CCTcontext to check.